qd_proto/dissectors/heartbeat.lua

```diff
@@ -0,0 +1,151 @@
+-- @file heartbeat.lua
+-- @brief Provides dissector for HEARTBEAT message type.
+package.prepend_path(Dir.global_plugins_path())
+local utils = require("qd_proto.utils")
+local dbg = require("qd_proto.dbg")
+
+local heartbeat = {}
+
+-- Bit flags content byte.
+-- Each bit determines which fields are present in this message.
+local content = {
+    EMPTY = 0,
+    TIME_MILLIS = 1,
+    TIME_MARK = 2,
+    DELTA_MARK = 4,
+    LAG_MARK = 8
+}
+
+-- List of HEARTBEAT fields.
+-- @note Some fields have a built-in type ProtoField.int64, but in fact the
+-- maximum value this fields is limited to 32-bits, this is done because
+-- compact_int type is used which can take up more then 32-bit, but value
+-- limited 32-bit (MSB bits define sizeof field).
+heartbeat.fields = {
+    -- Content byte.
+    content = ProtoField.uint8("qd.heartbeat.content", "Content", base.HEX),
+    -- Contest bit flags.
+    content_millis = ProtoField.bool("qd.heartbeat.content.has_millis",
+                                     "HAS_MILLIS", 4, nil, content.TIME_MILLIS),
+    content_time_mark = ProtoField.bool("qd.heartbeat.content.has_mark",
+                                        "HAS_TIME_MARK", 4, nil,
+                                        content.TIME_MARK),
+    content_delta_mark = ProtoField.bool("qd.heartbeat.content.has_delta",
+                                         "HAS_DELTA_MARK", 4, nil,
+                                         content.DELTA_MARK),
+    content_lag_mark = ProtoField.bool("qd.heartbeat.content.has_lag",
+                                       "HAS_LAG_MARK", 4, nil, content.LAG_MARK),
+    time_mills = ProtoField.int64("qd.heartbeat.time_mills",
+                                  "Time Milliseconds", base.DEC),
+    time_utc = ProtoField.absolute_time("qd.heartbeat.time_utc", "Time UTC",
+                                        base.UTC),
+
+    -- int32
+    time_mark = ProtoField.int64("qd.heartbeat.time_mark", "Time Mark", base.DEC),
+    -- int32
+    delta_mark = ProtoField.int64("qd.heartbeat.delta_mark", "Delta Mark",
+                                  base.DEC),
+    -- int32
+    lag_mark = ProtoField.int64("qd.heartbeat.lag_mark", "Lag Mark", base.DEC)
+}
+
+local function is_empty_content(content_byte)
+    return content_byte == content.EMPTY
+end
+
+local function has_time_millis(content_byte)
+    return (bit.band(content_byte, content.TIME_MILLIS) ~= 0) and true or false
+end
+
+local function has_time_mark(content_byte)
+    return (bit.band(content_byte, content.TIME_MARK) ~= 0) and true or false
+end
+
+local function has_delta_mark(content_byte)
+    return (bit.band(content_byte, content.DELTA_MARK) ~= 0) and true or false
+end
+
+local function has_lag_mark(content_byte)
+    return (bit.band(content_byte, content.LAG_MARK) ~= 0) and true or false
+end
+
+-- Dissect HEARTBEAT message.
```

Comment:
`Dissects the HEARTBEAT message`
It is worth applying the same requirements for describing parameters as above.